Sources/Sentry/SentryNSURLRequest.m

```diff
@@ -54,16 +54,17 @@ - (_Nullable instancetype)initStoreRequestWithDsn:(SentryDsn *)dsn
         [SentryLog logWithMessage:@"Using event->json attribute instead of serializing event" andLevel:kSentryLogLevelVerbose];
     } else {
         NSDictionary *serialized = [event serialize];
-        if (![NSJSONSerialization isValidJSONObject:serialized]) {
+        jsonData = [SentrySerialization dataWithJSONObject:serialized
+                                                   options:[SentrySDK.currentHub getClient].options.logLevel == kSentryLogLevelVerbose
+                                                           ? NSJSONWritingPrettyPrinted : 0
+                                                     error:error];
+        if (nil == jsonData) {
             if (error) {
+                // TODO: We're possibly overriding an error set by the actual code that failed ^
```

Comment:
same as above comment

Reply:
this problem was already there, I'm just pointing it out.
I would rather address these separately on a phase before GA as we did for Android